src/common/auditory_description.js

```diff
@@ -164,45 +164,556 @@ sre.AuditoryDescription.speechString = function(descrs, opt_separator) {
  * @private
  */
 sre.AuditoryDescription.toString_ = function(descrs, separator) {
-  return sre.Engine.getInstance().ssml ?
-      sre.AuditoryDescription.toSsmlString_(descrs, separator) :
-      sre.AuditoryDescription.toSimpleString_(descrs, separator);
+  switch (sre.Engine.getInstance().markup) {
+  case sre.Engine.Markup.SABLE:
+  case sre.Engine.Markup.SSML:
+  case sre.Engine.Markup.VOICEXML:
+    return sre.AuditoryDescription.toMarkupString_(descrs, separator);
+  case sre.Engine.Markup.ACSS:
+    return sre.AuditoryDescription.toAcssString_(descrs, separator);
+  case sre.Engine.Markup.NONE:
+  default:
+    return sre.AuditoryDescription.toSimpleString_(descrs, separator);
+  }
 };
 
```

Comment:
Refactor to audio module before 2.0 release.
Have audio format classes and a factory wrt. markup setting.